### PR TITLE
JBPM-6890: Adding the customTaskID as a unique identifier for an active

### PR DESCRIPTION
**Pull request by mcivantos-tribalyte**
Added processInstanceDescription as a businessId

### CODE REVIEW

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/TaskInstance.java [hunk 1]
```diff
@@ -64,6 +64,8 @@
     private Long parentId;
     @XmlElement(name="task-process-id")
     private String processId;
+    @XmlElement(name="process-instance-desc")
+    private String processInstanceDescription;
```
Comment (mswiderski) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/TaskInstance.java]: I don't like this one... it essentially means always a join to have this information which does not make much sense in my opinion. In other words process instance description does not belong to task instance so having it on TaskInstance type is misleading.

Comment (mcivantos-tribalyte) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/TaskInstance.java]: One of the custom queries register for the customer using the processInstanceDescription as a businessId and it's needed to find all the task related to this businesId (which is shared between several processes related to the same contract). If I don't add the processInstanceDescription to the TaskInstance model, how can we get the processInstanceDescription using the mapper I implemented and to which I added it?  (https://github.com/kiegroup/jbpm/pull/1146)

Comment (mswiderski) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/TaskInstance.java]: as described, you either create new type (TaskWithProcessInstance most likely better name would need to be used) and then map it here when using your additional jbpm service mappers. Or use the Raw mapper that will return array of arrays. Raw mapper is fine when KIE Server client is not used.

Comment (mcivantos-tribalyte) [reply kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/TaskInstance.java]: @mswiderski  I added a new type with the custom data used in my addition jbpm service mapper. Wdyt?

File: kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/TaskWithProcessDescription.java [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+        return outputData;
+    }
+
+    public void setOutputData(Map<String, Object> outputData) {
+        this.outputData = outputData;
+    }
+
+    @Override
+    public String toString() {
+        return "TaskInstance{" +
```
Comment (mswiderski) [inline kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/instance/TaskWithProcessDescription.java]: might be worth altering so it's not getting confused with the main TaskInstance

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/test/java/org/kie/server/services/jbpm/QueryDataServiceBaseTest.java [hunk 3]
```diff
@@ -40,15 +40,15 @@
 
     @Test
     public void testLoadDefaultQueryDefinitions() {
-       Date today = new Date();
-       KieServerEnvironment.setServerId(UUID.randomUUID().toString());
+        Date today = new Date();
+        KieServerEnvironment.setServerId(UUID.randomUUID().toString());
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm/src/test/java/org/kie/server/services/jbpm/QueryDataServiceBaseTest.java]: looks like some formatting issues ...

**Comment by mswiderski** [comment]
ok to test

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/QueryDataServiceIntegrationTest.java [hunk 4]
```diff
@@ -814,21 +814,24 @@ public void testGetTaskInstancesUsingUserTaskWithModificationMapper() throws Exc
...
+                "inner join PROCESSINSTANCELOG p on t.processinstanceid = p.processinstanceid " +
+                "inner join TASKVARIABLEIMPL d on t.id=d.taskid");
         query.setTarget("CUSTOM");
         
         try {
 
             queryClient.registerQuery(query);
 
-            List<TaskInstance> tasks = queryClient.query(query.getName(), QueryServicesClient.QUERY_MAP_TASK_WITH_MODIF, 0, 10, TaskInstance.class);
+            List<TaskWithProcessDescription> tasks = queryClient.query(query.getName(), QueryServicesClient.QUERY_MAP_TASK_WITH_MODIF, 0, 10, TaskWithProcessDescription.class);
```
Comment (MarianMacik) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/QueryDataServiceIntegrationTest.java]: Don't we lose coverage of a scenario with TaskInstance this way?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/ConvertUtils.java [hunk 5]
```diff
@@ -376,23 +378,23 @@ public static TaskInstance convertToTask(UserTaskInstanceDesc userTask) {
         return instance;
     }
     
-    public static TaskInstanceList convertToTaskInstanceListPO(Collection<UserTaskInstanceWithPotOwnerDesc> instances) {
+    public static TaskWithProcessDescriptionList convertToTaskInstanceListPO(Collection<UserTaskInstanceWithPotOwnerDesc> instances) {
```
Comment (MarianMacik) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/ConvertUtils.java]: TaskInstanceList will be no longer used?

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/QueryDataServiceIntegrationTest.java (hunk 4 above)
Comment (MarianMacik) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/QueryDataServiceIntegrationTest.java]: Do we test scenario with TaskInstance entity somewhere else? Wouldn't it be better to have a separate test for TaskWithProcessDescription?

**Comment by mswiderski** [comment]
Jenkins retest this

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/QueryDataServiceIntegrationTest.java (hunk 4 above)
Comment (mswiderski) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/QueryDataServiceIntegrationTest.java]: no, it's for special type of mapper so it's expected to use this new type

File: kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/ConvertUtils.java (hunk 5 above)
Comment (mswiderski) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/ConvertUtils.java]: it's used for all other types but not the PO customised one

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/QueryDataServiceIntegrationTest.java (hunk 4 above)
Comment (mswiderski) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/QueryDataServiceIntegrationTest.java]: yes, TaskInstance is used whenever task are returned from other methods. Moreover, it's really driven by the last argument in the query method

Comment (MarianMacik) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/QueryDataServiceIntegrationTest.java]: Thanks for all answers.

**Comment by mswiderski** [comment]
@mcivantos-tribalyte please have a look at failures as they seem to be related to changes

**Comment by mcivantos-tribalyte** [comment]
Take into account that this PR is related to https://github.com/kiegroup/jbpm/pull/1148

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by mswiderski** [comment]
Jenkins retest this

**Pull request merged**